zgw/notificaties-api/src/main/kotlin/com/ritense/notificatiesapi/web/rest/NotificatiesApiResource.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.notificatiesapi.web.rest
+
+import com.ritense.notificatiesapi.event.NotificatiesApiNotificationReceivedEvent
+import com.ritense.notificatiesapi.service.NotificatiesApiService
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RequestMapping
+
+@RequestMapping(value = ["/api"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class NotificatiesApiResource(
+    private val notificatiesApiService: NotificatiesApiService
+) {
+
+    @PostMapping("/v1/notificaties/send")
+    fun handleNotification(
+        @RequestBody notification: NotificatiesApiNotificationReceivedEvent,
+        @RequestHeader("Authorization") authHeader: String?
+    ): ResponseEntity<Void> {
+        return if (authHeader != null) {
+            try {
+                notificatiesApiService.findAbonnementSubscription(authHeader)
+                notificatiesApiService.handle(notification)
+                ResponseEntity.noContent().build()
+            } catch (ex: RuntimeException) {
+                ResponseEntity.status(HttpStatus.UNAUTHORIZED).build()
```

Comment:
A RuntimeException is should not result in an UNAUTHORIZED message. Because the notification can be handled in may different listeners that can all throw an exception.

Exception handeling is e big reason of why we are rewriting the existing connector to a new plugin. In the previous connector, exceptions were often not visible.

Also the exception message is being swallowed here. The exception message with stacktrace should be returned by this endpoint. The exception will then end up in the logs of OpenNotificaties. This helps with debugging